articles/quickstart/webapp/nodejs-ea/01-login.md

```diff
@@ -0,0 +1,110 @@
+# Using NPM
+npm install express-openid-connect express-session
+
+# Using Yarn
+yarn add express-openid-connect express-session
+```
+
+### Handling server responses
+Your application will need to parse url-encoded payloads sent back from the Auth0 server.  Express provides a middleware for this `express.urlencoded`. You might already be using `body-parser`, this will work too.
+
+```js
+const express = require('express');
+const app = express();
+
+app.use(express.urlencoded({
+  extended: false
+}));
+```
+
+### User sessions
+In order to keep the user logged in across requests you will need to use session middleware.  The session will allow you to recognize the user on concurrent requests. In this example, you will use `express-session` which stores the session in memory.
+
+```js
+const session = require('express-session');
+
+app.use(session({
+  secret: 'this should be a secret',
+  resave: true,
+  saveUninitialized: false
+}));
+```
+
+### Authentication
+The last part of integrating Auth0 into your application will be using the `auth` middleware supplied to you from the Auth0 library `express-opeinid-connect`.  View the official [Api Documentation](https://github.com/auth0/express-openid-connect/blob/4374c0502d4aedf8b3975e7c5f26929b305b32f6/API.md) for configuration options and information.
+
+```js
+const { auth } = require('express-openid-connect');
+
+app.use(auth({
+  required: false,
+  auth0Logout: true,
+  baseURL: 'http://localhost:3000',
+  issuerBaseURL: 'https://${account.namespace}',
+  clientID: '${account.clientId}'
+}));
+```
+
+## Protect a Route
+Now that your application has the required middleware to integrate Auth0 in your application, you can start protecting routes.
+
+Add the `requiresAuth` middleware supplied from `express-openid-connect` for routes that require authentication.  Any route using this middleware will check for a valid user session and, if one does not exist, it will redirect the user to log in.
+
+```js
+const { requiresAuth } = require('express-openid-connect');
+
+app.get('/profile', requiresAuth(), (req, res) => {
+  // when a user is logged in you can access the user object via `req.openid.user`
+  res.send(JSON.stringify(req.openid.user));
+});
+```
+
+## Login and Logout
+### Login
+To log a user in you have 3 options:
+- Redirect to the `/login` route registered by the `express-oidc-connect.auth` middleware.
+- Use the `express-oidc-connect.requiresAuth` middleware to protect a route, as described above.
+- When initializing the `express-oidc-connect.auth` middleware, pass in `required: true` to force authentication on all routes.
+
+::: note
+If you are testing your application locally, you can login by visiting `http://localhost:3000/login`
+:::
+
+### Logout
+To log a user out, redirect them to the `/logout` route registered by the `express-oidc-connect.auth` middleware.
+
+::: note
+If you are testing your application locally, you can logout by visiting `http://localhost:3000/logout`
```

Comment:
```suggestion
If you are testing your application locally, you can logout by visiting [`localhost:3000/logout`](http://localhost:3000/logout)
```

I would combine this with the sentence above instead of calling it out (IMHO)